pom.xml

```diff
@@ -1024,7 +1025,7 @@
       <dependency>
         <groupId>org.apache.hadoop</groupId>
         <artifactId>hadoop-minikdc</artifactId>
-        <version>${hadoop.version}</version>
+        <version>${minikdc.version}</version>
```

Comment:
I don't think we can mix and match Hadoop versions this way. I think generally speaking Hadoop 3.x is what we'd tell people to use for JDK 11 compatibility. But if the other change to refreshKrb5Config is needed and compatible, we can make that change.

Reply:
Yes. Agree with @srowen.
Could we fix this issue by changing the `default_tkt_enctypes` and `default_tgs_enctypes`:
https://www.oracle.com/technetwork/java/javase/11-relnote-issues-5012449.html#JDK-8014628

Reply:
Please also see https://issues.apache.org/jira/browse/KAFKA-7338

Reply:
@srowen @wangyum @dongjoon-hyun 
I checked the kerberos code that when client call a request `KrbTgsReq/KrbTktReq `, it will first check if there are config `default_tkt_enctypes` and `default_tgs_enctypes`in `krb5.conf`,  if not, then use `jdk buildin enctypes`. 
And  i have checked the build in ecntypes:
```
jdk8
  static {
    DEBUG = Krb5.DEBUG;
    initStatic();
    BUILTIN_ETYPES = new int[]{18, 17, 16, 23, 1, 3};
    BUILTIN_ETYPES_NOAES256 = new int[]{17, 16, 23, 1, 3};
  }

jdk11
 static {
    DEBUG = Krb5.DEBUG;
    initStatic();
    BUILTIN_ETYPES = new int[]{18, 17, 20, 19, 16, 23, 1, 3};
    BUILTIN_ETYPES_NOAES256 = new int[]{17, 19, 16, 23, 1, 3};
  }
```
In server side , they both use `aes128-cts-hmac-sha1-96` as first choice. 
But MiniKdc don't have API for us to config these. So rewrite and refresh it.